modular_citadel/code/modules/fish/fishtank.dm

```diff
@@ -0,0 +1,676 @@
+			examine_message += "The fish seem happy! "
+		else if(food_level == 10)
+			examine_message += "There is a solid layer of fish food at the top. "
+
+	//Report the number of harvestable eggs
+	if(egg_count)								//Don't bother if there isn't any eggs
+		examine_message += "<br>There are [egg_count] eggs able to be harvested! "
+
+	examine_message += "<br>"
+
+	//Report the number and types of live fish if there is water in the tank
+	if(fish_count == 0)
+		examine_message += "\The [src] doesn't contain any live fish. "
+	else
+		//Build a message reporting the types of fish
+		var/fish_num = fish_count
+		var/message = "You spot "
+		while(fish_num > 0)
+			var/datum/fish/fish_type = fish_list[fish_num]
+			var/fish_name = fish_type.fish_name
+			if(fish_count > 1 && fish_num == 1)	//If there were at least 2 fish, and this is the last one, add "and" to the message
+				message += "and "
+			message += "\an [fish_name]"
+			fish_num --
+			if(fish_num > 0)					//There's more fish, add a comma to the message
+				message +=", "
+		message +="."							//No more fish, end the message with a period
+		//Display the number of fish and previously constructed message
+		examine_message += "\The [src] contains [fish_count] live fish. [message] "
+
+	examine_message += "<br>"
+
+	//Report lid state for tanks and wall-tanks
+	if(has_lid)									//Only report if the tank actually has a lid
+		//Report lid state
+		if(lid_switch)
+			examine_message += "The lid is closed. "
+		else
+			examine_message += "The lid is open. "
+
+	examine_message += "<br>"
+
+	//Report if the tank is leaking/cracked
+	if(water_level > 0)							//Tank has water, so it's actually leaking
+		if(leaking == 1) examine_message += "\The [src] is leaking."
+		if(leaking == 2) examine_message += "\The [src] is leaking profusely!"
+	else										//No water, report the cracks instead
+		if(leaking == 1) examine_message += "\The [src] is cracked."
+		if(leaking == 2) examine_message += "\The [src] is nearly shattered!"
+
+
+	//Finally, report the full examine_message constructed from the above reports
+	to_chat(user, "[examine_message]")
+	return examine_message
+
+//////////////////////////////
+//		ATACK PROCS			//
+//////////////////////////////
+
+/obj/machinery/fishtank/attack_animal(mob/living/simple_animal/M as mob)
+	if(istype(M, /mob/living/simple_animal/pet/cat))
+		if(M.a_intent == INTENT_HELP)							//Cats can try to fish in open tanks on help intent
+			if(lid_switch)									//Can't fish in a closed tank. Fishbowls are ALWAYS open.
+				M.visible_message("[M.name] stares at into \the [src] while sitting perfectly still.", "The lid is closed, so you stare into \the [src] intently.")
+			else
+				if(fish_count)								//Tank must actually have fish to try catching one
+					M.visible_message("[M.name] leaps up onto  \the[src] and attempts to fish through the opening!", "You jump up onto \the [src] and begin fishing through the opening!")
+					spawn(10)
```

Comment:
what i find silly is this doesn't even stun the cat ora nything. also, no spawn.